### ISSUE TITLE
Right click "connect-to" other node

### ISSUE THREAD

**Issue by keyserj** [amelioro/ameliorate#632]
### Describe your issue

Connecting to a node across the map is annoying:

https://github.com/user-attachments/assets/275ff969-66f9-4983-a78a-7c6f98afae63

### Solution you'd like

Right click "connect-to" other node.

- ideally would come with a node-search component and allow just clicking on another node (related #376 #381)
- maybe "connect-to-parent" vs "connect-to-child"?

**Comment by Ferbalot** [comment]
I also think that (at least when you zoom out) the connection circles to begin/end node connections should be much bigger to make it easier to connect nodes while very zoomed out. I think this would be much easier for me than a right click "connect-to" menu in most cases.

**Comment by keyserj** [comment]
@Ferbalot that seems like it could help in some cases, but if you have to zoom out too far, it seems like the node text size would still make it hard to identify which nodes you want to connect.

What do you think about start a "connecting..." action if you click on a handle rather than drag, which overlays the node at the bottom of the screen and says something like "click another handle to create a connection"? Then you can pane around as you want without having to zoom out to keep both nodes on the screen at the same time.

One issue with this "connecting..." idea is that clicking the handle on mobile is currently used for showing hidden neighbors in that handle's direction. Somehow these two features would have to avoid conflict.

(I retitled this issue to be based on the concern rather than on the solution, since right-clicking may not be the best solution.)

**Comment by Ferbalot** [comment]
@keyserj I was thinking that in most cases, either you wouldn't have to zoom out too far or the structure would make it clear enough for you to know what node you're looking to connect to. But thinking about it more, I do think I underestimated the speed at which you could use the search system to connect to another node.

For some reason I really feel like dragging would be better than the "connecting..." action in many cases (probably because I'm used to it) but I tentatively think this feeling is unjustified and the "connecting..." action would be better in almost all cases and greatly reduce the usefulness of bigger circles. I still think bigger circles would help a decent amount since they seem really small now, but I'm less confident it's worthwhile.

I'm likely just too inexperienced to recognize the usefulness of seeing hidden neighbors, but with my current understanding of their usefulness, I would tentatively think the options to show hidden nodes in a direction seems better relegated to a side menu or double-tap or something instead of that.

**Comment by keyserj** [comment]
The UX for show/hiding neighbors via the handle is a bit awkward, so I don't actually use it too much 😅 . Adding #657 would make it almost entirely obsolete, except that the hidden neighbor functionality allows you to show individual neighbors, where collapse/expand seems like it'd just do all of them. Perhaps, if the individual control seems worth keeping, the collapse/expand button could be right-clicked to show/hide individual neighbors.
